linux_os/guide/system/accounts/accounts-restrictions/password_expiration/accounts_password_set_max_life_existing/bash/shared.sh

```diff
@@ -7,11 +7,14 @@
 {{{ bash_instantiate_variables("var_accounts_maximum_age_login_defs") }}}
 
 {{% if product in ["sle12", "sle15"] %}}
-usrs_max_pass_age=( "$(awk -F: '$5 > $var_accounts_maximum_age_login_defs || $5 == "" {print $1}' /etc/shadow)" )
-for i in "${usrs_max_pass_age[@]}"
-do
-  passwd -q -x $((var_accounts_maximum_age_login_defs)) $i
-done
+while IFS= read -r line; do
```

Comment:
Hello @marcusburghardt
1/
I tried to use chage - but I had issues related to PAM (i.e. PAM blocked me to apply this change). With passwd I did not have these issues. I think that RedHat is not protected from this issue also. 
2/
The second issue which I had was to apply the change via awk. When I did cat /etc/shadow - I had an account which had to be changed, but it remained unchanged i.e. awk did not take it in consideration. Now with the current approach I was able to apply changes for all accounts.  
Have a nice day
Rumen

Reply:
> Hello @marcusburghardt 1/ I tried to use chage - but I had issues related to PAM (i.e. PAM blocked me to apply this change). With passwd I did not have these issues. I think that RedHat is not protected from this issue also.

It is fine to use `passwd` for `sle` products. My point was to put only this inside jinja conditionals since the list of users can be shared among other distros.

> 2/ The second issue which I had was to apply the change via awk. When I did cat /etc/shadow - I had an account which had to be changed, but it remained unchanged i.e. awk did not take it in consideration. Now with the current approach I was able to apply changes for all accounts. Have a nice day Rumen

Could you provide an example, please? I believe the `awk` command can be easily extended to cover more cases.

Reply:
Hello @marcusburghardt 
The awk approach is not working correctly. Please see my message with attached files. The test_max_with_passwd.txt cover the case where we have a loop based on awk, and we do change with passwd. Only two accounts were changed! 
With my approach I scan /etc/shadow ones (I do not collect accounts) and if during the scan there is a need of a change I do it.